pcdsdevices/ladm.py

```diff
@@ -0,0 +1,423 @@
+            else:
+                print(f"Moving x1 to {x1_var} and x2 to {x2_var}")
+                self.x1.mvr(x1_var)
+                self.x2.mvr(x2_var)
+                self.x1.wait()
+                self.x2.wait()
+                print("Done")
+                print(f"Moving z to {z_var}")
+                self.z.mvr(z_var - z_now)
+                print("Done")
+        except KeyboardInterrupt:
+            self.stop()
+
+    def mvrV(self, y):
+        """ ladm move vertically  """
+        try:
+            self.y1.mvr(y)
+            self.y2.mvr(y)
+        except KeyboardInterrupt:
+            self.stop()
+
+    def wmX(self):
+        x1_var = self.x1.wm()
+        x2_var = self.x2.wm()
+        z_var = self.z.wm()
+        x_x1, x_x2, x_z = MotorsTox(x1_var, x2_var, z_var)
+        db_x1 = EpicsSignal(self.x1.prefix+'.RDBD', name='db_x1').get()
+        db_x2 = EpicsSignal(self.x2.prefix+'.RDBD', name='db_x2').get()
+        tolerance = db_x1 + db_x2
+        if (abs(x_x1 - x_x2) <= tolerance):
+            z_theo = xToz(x_x1)
+            db_z = EpicsSignal(self.z.prefix+'.RDBD', name='db_z').get()
+            if abs(z_var - z_theo) < 2 * db_z:
+                return x_x1
+            else:
+                return np.nan
+        else:
+            return np.nan
+
+    def ca_theta(self, cal_theta, samz_offset=0):
+        """
+        calculation relative x and z postion at certain theta
+        ca_theta(theta,samz_offset(downstream offset / mm))
+        """
+        cal_thetarad = cal_theta * np.pi/180
+        cal_x1 = ((r - samz_offset * alpha_cos) *
+                  np.sin(cal_thetarad) / (np.sin(alpha_r) *
+                  np.sin(alpha_r + cal_thetarad)))
+        cal_x2 = ((R - samz_offset * alpha_cos) *
+                  np.sin(cal_thetarad)/(np.sin(alpha_r) *
+                  np.sin(alpha_r + cal_thetarad)))
+        cal_dz = ((r - samz_offset * alpha_cos) /
+                  np.sin(alpha_r) - cal_x1 *
+                  np.sin(alpha_r)/np.sin(cal_thetarad))
+        caTable = PrettyTable()
+        caTable.field_names = ["Theta", "x1", "x2", "delta z",
+                               "sample z offset"]
+        caTable.add_row([cal_theta, cal_x1, cal_x2, cal_dz, samz_offset])
+        print(caTable)
+        return cal_theta, cal_x1, cal_x2, cal_dz, samz_offset
+
+    def _setX(self, value):
+        x1_var = xTox12(value)
+        x2_var = xTox12(value)
+        z_var = xToz(value)
+        self.x1.set(x1_var)
+        self.x2.set(x2_var)
+        self.z.set(z_var)
+
+    def _set_lowlimX(self, value):
+        self.__lowlimX = value
+
+    def _set_hilimX(self, value):
+        self.__hilimX = value
+
+    def _get_lowlimX(self):
+        return self.__lowlimX
+
+    def _get_hilimX(self):
+        return self.__hilimX
+
+    def stop(self):
+        self.x1.stop()
+        self.x2.stop()
+        self.y1.stop()
+        self.y2.stop()
+        self.z.stop()
+
+    def __call__(self):
```

Comment:
Nitpick: We probably want to use the standard `format_status_info` method that displays device information in ipython.  That way you don't actually have to write the parentheses `()` when you want to see the status info.

More concisely, I think if we define `LADM.format_status_info` we should just be able to type LADM in the ipython terminal to see the status info